### PR TITLE
Fixed #4: added compile directive

### LINKED ISSUE #4: Add compile directive
Very useful when pulling HTML (static or generated with a REST transform, potentially containing angular instructions) that you want to inject into your view:

```javascript
// Copied from https://docs.angularjs.org/api/ng/service/$compile
angular.module('sample.create')
  .directive('compile', function($compile) {
    'use strict';

    // directive factory creates a link function
    return function(scope, element, attrs) {
      scope.$watch(
        function(scope) {
           // watch the 'compile' expression for changes
          return scope.$eval(attrs.compile);
        },
        function(value) {
          // when the 'compile' expression changes
          // assign it into the current DOM
          element.html(value);

          // compile the new DOM and link it to the current
          // scope.
          // NOTE: we only compile .childNodes so that
          // we don't get into infinite loop compiling ourselves
          $compile(element.contents())(scope);
        }
      );
    };
  });
```

Since it is being removed from our slush template, I'd be grateful to give it a new home here so I can still easily make use of it..

### PR DESCRIPTION
**Pull request by grtjn**
Fixes #4

### CODE REVIEW

**Comment by joemfb** [comment]
Thanks, this looks good!

One change request: can you remove the second argument to `angular.module()`? I think that's duplicating the initialization of `ml.common`.

File: src/compile.directive.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copied from https://docs.angularjs.org/api/ng/service/$compile
+angular.module('ml.common', [], function($compileProvider) {
```
Comment (joemfb) [inline src/compile.directive.js]: Please remove the second argument here. See https://docs.angularjs.org/guide/module#creation-versus-retrieval for details.

**Comment by grtjn** [comment]
You are right, I made a change, after testing in Splunk: http://plnkr.co/edit/Opiq0Et2xQAIPMcDU71V?p=preview. Looks much simpler as well, the $compileProvider didn't seem to be necessary.

**Comment by joemfb** [comment]
Awesome! Finally, can you squash your commits down to one?

**Comment by grtjn** [comment]
If you tell me how.. Is it very important? :-P

**Comment by joemfb** [comment]
It's just nice to keep the history clean with atomic commits.

    git checkout 4-compile-directive
    git rebase -i HEAD~2

replace "pick" with "squash" in second line.
save
when prompted, edit / combine commit messages

    git push origin 4-compile-directive -f

**Comment by grtjn** [comment]
`squash` didn't seem enough, I used `fixup` instead. Anyhow, looks good now!

**Comment by joemfb** [comment]
Looks great!

**Pull request merged**